### PR TITLE
Refactor TensorFlow operators

### PR DESCRIPTION
**Pull request by stu1130**
PR 1 Refactor TensorFlow operators
* There are several complicate ops implementation, will re-implement those using new arch

### CODE REVIEW

File: tensorflow/tensorflow-engine/src/main/java/ai/djl/tensorflow/engine/TfNDArray.java [hunk 1]
```diff
@@ -326,27 +300,8 @@ public void copyTo(NDArray ndArray) {
...
-                tf.gather(
-                                getOperand(),
-                                tf.squeeze(
-                                        tf.where(((TfNDArray) index).getOperand()),
-                                        Squeeze.axis(Collections.singletonList((long) 1))),
-                                tf.constant(axis))
-                        .asTensor()) {
-            return new TfNDArray(manager, tensor);
-        }
+        // FIXME re-implement
```
Comment (lanking520) [inline tensorflow/tensorflow-engine/src/main/java/ai/djl/tensorflow/engine/TfNDArray.java:303]: // FIXME: re-implement

File: tensorflow/tensorflow-engine/src/main/java/ai/djl/tensorflow/engine/TfNDArray.java [hunk 2]
```diff
@@ -573,26 +541,27 @@ public NDArray norm(int ord, int[] axes, boolean keepDims) {
...
+        opBuilder.addInput(((TfNDArray) manager.create(axes)).getHandle().asOutput());
+        opBuilder.setAttr("keep_dims", keepDims);
+        try (Tensor tensor = opBuilder.build().output(0).asTensor()) {
             return new TfNDArray(manager, tensor);
         }
     }
 
     /** {@inheritDoc} */
     @Override
+    @SuppressWarnings("unchecked")
```
Comment (frankfliu) [inline tensorflow/tensorflow-engine/src/main/java/ai/djl/tensorflow/engine/TfNDArray.java:556]: Do we need this suppression?

File: tensorflow/tensorflow-engine/src/main/java/ai/djl/tensorflow/engine/TfNDArray.java [hunk 3]
```diff
@@ -192,7 +163,7 @@ public NDArray stopGradient() {
                     "DataType mismatch, Required double" + " Actual " + getDataType());
         }
         double[] result = new double[(int) getShape().size()];
-        try (Tensor tensor = operand.asTensor()) {
+        try (Tensor tensor = getHandle().asTensor()) {
```
Comment (frankfliu) [inline tensorflow/tensorflow-engine/src/main/java/ai/djl/tensorflow/engine/TfNDArray.java:166]: We are still using Tensor, can we bypass Tensor and use TF_Tensor directly?

Comment (stu1130) [reply tensorflow/tensorflow-engine/src/main/java/ai/djl/tensorflow/engine/TfNDArray.java:166]: Sure, that's why I am doing now. This is just the first part of huge PR

**Pull request merged**